### PR TITLE
deprecating nstemplateset fields 

### PR DESCRIPTION
**Pull request by tinakurian**
removing revision, type and tiername from nstemplate set. 

Jira: https://issues.redhat.com/browse/CRT-503
related PRs:
api - https://github.com/codeready-toolchain/api/pull/143
host - https://github.com/codeready-toolchain/host-operator/pull/212

### CODE REVIEW

**Comment by MatousJobanek** [comment]
@tinakurian could you please add the replace clause for api and common repos to the `go.mod`? To know that we don't break anything with this change

**Comment by tinakurian** [comment]
> @tinakurian could you please add the replace clause for api and common repos to the `go.mod`? To know that we don't break anything with this change

fixed here: dfa60c0f8d064b4678cd233be3cd1c54435e9afc

**Comment by MatousJobanek** [comment]
> fixed here: dfa60c0

@tinakurian but you have added replacement only for api repo, not for common

**Comment by tinakurian** [comment]
> > fixed here: [dfa60c0](https://github.com/codeready-toolchain/member-operator/commit/dfa60c0f8d064b4678cd233be3cd1c54435e9afc)
> 
> @tinakurian but you have added replacement only for api repo, not for common

@MatousJobanek i didn't update common because none of the funcs changed in common are used in member-operator. however, i've updated anyway: add0e0792a295634113d3efaf52b26236c6b6504

**Comment by MatousJobanek** [comment]
> i didn't update common because none of the funcs changed in common are used in member-operator. however, i've updated anyway: 

I know, but as I have already written in your e2e PR - this is a big change in our API and it's good to be 100% sure that we won't break anything. In addition, this PR is changing the dependencies anyway, so why not update the common repo too :-)

File: go.mod [hunk 1]
```diff
@@ -63,4 +63,8 @@ replace (
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200204173128-addea2498afe // avoids case-insensitive import collision: "github.com/googleapis/gnostic/openapiv2" and "github.com/googleapis/gnostic/OpenAPIv2"
 )
 
+replace github.com/codeready-toolchain/api v0.0.0-20200525084630-00f2f69b8984 => github.com/tinakurian/api v0.0.0-20200528175500-de4844132cd6
+
+replace github.com/codeready-toolchain/toolchain-common v0.0.0-20200528070449-7c8b2f895b25 => github.com/tinakurian/toolchain-common v0.0.0-20200601144142-3440b7317af0
```
Comment (MatousJobanek) [inline go.mod:68]: Reminder to remove these two lines as soon as the two PRs are merged

File: go.mod [hunk 2]
```diff
@@ -63,4 +63,8 @@ replace (
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200204173128-addea2498afe // avoids case-insensitive import collision: "github.com/googleapis/gnostic/openapiv2" and "github.com/googleapis/gnostic/OpenAPIv2"
 )
 
+replace github.com/codeready-toolchain/api v0.0.0-20200525084630-00f2f69b8984 => github.com/tinakurian/api v0.0.0-20200528175500-de4844132cd6
```
Comment (xcoulon) [inline go.mod:66]: reminder ;)

**Pull request merged**